third_party/nvfuser/csrc/index_compute.h

```diff
@@ -320,12 +321,20 @@ class Index {
   // get the strides of a tensor used for the index lowering
   static std::vector<Val*> getStrides(const TensorView* tv);
 
-  // get the root indices of a tensor used for the index lowering
-  static std::vector<Val*> getRootIndices(
+  // get the root indices of a consumer tensor
+  static std::vector<Val*> getConsumerRootIndices(
       const TensorView* tv,
       const std::vector<kir::ForLoop*>& loops,
       const IndexFromIdGraph& index_from_id_graph);
 
+  // get the root indices of a producer tensor
+  static std::vector<Val*> getProducerRootIndices(
+      TensorView* producer,
+      const TensorView* consumer,
+      const std::vector<kir::ForLoop*>& loops,
+      const std::unordered_set<kir::ForLoop*>& rotated_loops,
```

Comment:
I think it would be good to try and keep indexing as self contained as possible. Having these arguments passed in like this make me worried about trying to refactor indexing. Can revisit later.